Modified and reorganized the VSM docs

Preeticp opened the pull request:
Modified the Readme, Contributing and Maintainers doc. Added a new doc - Available commands and Exit codes. This takes care of issue #77, and part of #117.

hferentschik commented:
A couple of things: 
- I am all for clean and lean root directories, but some files do belong there. For example _CONTRIBUTING.md_ has a special meaning when places in the root directory. The content of the file is displayed to the user when a pull request is created in order to help him fulfilling project guidelines - https://github.com/blog/1184-contributing-guidelines. So at least  _CONTRIBUTING.md_ should stay in the root
- Personally I am not a big fan of the `MAINTAINERS` file in general (and would actually prefer to remove it, GitHub maintain this information for you anyways and it is in the git logs as well), but if you have such a file it belongs into the root imo
- I like the idea of splitting up the docs, but why putting the important things like which commands exist, what they do and which exit code they have into a sub-page (which does not even seem to be linked from the README)?
- Please add a table of contents with links to the sub-sections. It helps so much navigating in longer documents. It is a bit of a pain to do manually in markdown, but some editors (eg Sublime Text) can do it automatically for you

hferentschik commented on docs/Available commands and exit codes.md:
```diff
@@ -0,0 +1,24 @@
+# vagrant-service-manager commands and exit codes
+
+Commands                                                | Explanations
+--------------------------------------------------------|-----------------------------------------
```
Why putting this essential information into a sub page instead of the main README? There are other things which can be split out from there, if one wants to trim down the README.

hferentschik commented on CONTRIBUTING.md:
```diff
@@ -1,31 +0,0 @@
-# Contributing to vagrant-service-manager Plugin
```
-1 for moving this file out of the root directory

Preeticp commented:
Agree with you @hferentschik . Thanks for your suggestions. I have updated the VSM docs as per your suggestions.
I have moved the Contributing and Maintainers documents back to root. I have also moved the commands and exit codes to the Readme, where a user can find it upfront. 
Some sections of the Readme have been moved to the Maintainers and Contributing as per relevance. Most of the quick start will be moved to the ADB Usage document, to avoid duplication, so the Readme has been shortened to an extent.

bexelbie commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,32 @@
+You can submit issues with respect to the vagrant-service-manager plugin [here](https://github.com/projectatomic/vagrant-service-manager/issues/new).Make sure you include all the relevant details pertaining the issue.
+Before submitting new issues, it is suggested to check [all existing issues](https://github.com/projectatomic/vagrant-service-manager/issues) in order to avoid duplication.
+
+
```
2 things:
1. Should we move this to the README or at least put in a pointer?
2. Should we add something like:

The vagrant service-manager plugin works closely with the Atomic Developer Bundle (ADB) and the adb-utils RPM.  You may wish to review the issues in both repos as well.

With appropriate links

bexelbie commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,32 @@
-* All changes must include an entry in the `CHANGELOG.md` in the
-  *unreleased* section describing the change
+* All changes must include an entry in the [Changelog document](https://github.com/projectatomic/vagrant-service-manager/blob/master/CHANGELOG.md) in the
+  *Unreleased* section describing the change.
```
add: Please include your github userid in the Changelog.

bexelbie commented on MAINTAINERS:
```diff
@@ -1,3 +1,15 @@
+
+## How to release the vagrant-service-manager plugin
+
+1. Put the gemfile in pkg/ with `rake build`
```
Should this be "Build the gemfile in pkg/ ..." @LalatenduMohanty @navidshaikh

hferentschik replied:
or: "Create gem file under _pkg_ by running `rake build` ;-)
_put_ is misleading in this context indeed.

navidshaikh replied:
"create gem file under `pkg` directory by running `rake build` "

LalatenduMohanty replied:
Do we really need "## How to release the vagrant-service-manager plugin" section in  MAINTAINERS  file. IMO it does not belong here.

bexelbie replied:
I'd like it to land here, pending further reorg of the docs.  We need this somewhere.

bexelbie commented on README.md:
```diff
@@ -64,76 +46,44 @@ producing complex, multi-container applications.
 
-Install the plugin using:
+We welcome your input. Refer to the [contributing guidelines](https://github.com/projectatomic/vagrant-service-manager/blob/master/CONTRIBUTING.md) for information on how to contribute to this plugin.
 
```
The link to the issues notes in CONTRIBUTING could go here

navidshaikh commented on README.md:
```diff
@@ -64,76 +46,44 @@ producing complex, multi-container applications.
         # run following command to configure your shell:
```
output of the `vagrant service-manager env docker` needs to be updated to match the latest output. 
Also value of `DOCKER_CERT_PATH` needs to have a dummy path (currently its showing my system's path)

bexelbie replied:
Can you provide those for @Preeticp  - her machine got borked so she may have a delay before she can run a test run

navidshaikh replied:
Here is the sample output with latest bits in master and dummy path for value for `DOCKER_CERT_PATH`

```
$ vagrant service-manager env docker
# Set the following environment variables to enable access to the
# docker daemon running inside of the vagrant virtual machine:
export DOCKER_HOST=tcp://172.28.128.7:2376
export DOCKER_CERT_PATH=/foo/bar/.vagrant/machines/default/virtualbox/docker
export DOCKER_TLS_VERIFY=1
export DOCKER_API_VERSION=1.21
# run following command to configure your shell:
# eval "$(vagrant service-manager env docker)"

```

navidshaikh commented on README.md:
```diff
@@ -1,56 +1,38 @@
-6. Run the plugin to get environment variables and certificates:
+	**Note:** When the vagrant-service-manager plugin is loaded and a box is started using the VirtualBox provider, the user needs to add a routable non NAT network interface declaration in the Vagrantfile. If the user does not provide a network declaration in the Vagrantfile, a private DHCP network is added by default and a warning is displayed.
+
+4. Run the plugin to get environment variables and certificates:
```
@bexelbie : for this PR, do we want to have just one sample run of command? and we plan to include the different examples of command execution with elaboration?

bexelbie replied:
I am good with a sample run.  At a later time we can decide if we need more.

LalatenduMohanty commented on README.md:
```diff
@@ -1,56 +1,38 @@
+[ADB](https://github.com/projectatomic/adb-atomic-developer-bundle) should download the relevant [Vagrantfile from the repository](https://github.com/projectatomic/adb-atomic-developer-bundle/tree/master/components/centos). For further details and to use custom vagrant files designed for specific use cases, refer to the [usage documentation](https://github.com/projectatomic/adb-atomic-developer-b ...
 
-5. Start the ADB using `vagrant up`. Users of the ADB may wish to consult the
+3. Start the vagrant box or ADB using `vagrant up`. For detailed instructions consult the
```
It should be `Start the ADB  Vagrant box using` because the installation documentation link is for ADB.

Preeticp replied:
The previous step says, box of your choice, so the next step in that context says, start the vagrant box (of your choice) or ADB. What I can do  is, in the next sentence, add, - Users of ADB may consult the installation documentation.

LalatenduMohanty replied:
yes I have seen that before. But does VSM have an use case  outside ADB and CDK ? Then why confuse users saying that it can be used outside of ADB/CDK @bexelbie @navidshaikh

navidshaikh replied:
We have guest capability to detect ADB and CDK and act accordingly. Its a bit far now for VSM to be generic for other boxes as well. I suggest we mention only about ADB / CDK.

bexelbie commented:
@Preeticp please rebase/update this PR.

Preeticp commented:
Rebased and updated the PR

navidshaikh commented:
@bexelbie : This needs to be in master. This should help @Preeticp move forward quickly by avoiding the conflicts in other PRs. Since this is a doc PR, I would like you to give a master ack before it gets merged. My non master ACK to this PR! :-) LGTM!

bexelbie commented:
LGTM - though commands section is out of date - will open a new issue
